src/main/java/com/github/manolo8/darkbot/gui/titlebar/ExtraButton.java

```diff
@@ -31,12 +34,44 @@
     private static final Set<ExtraMenuProvider> EXTRA_DECORATIONS = new LinkedHashSet<>();
     private static final Lazy<Void> clean = new Lazy.NoCache<>();
 
-    public static void setExtraDecorations(Stream<ExtraMenuProvider> provider) {
+    public static void setExtraDecorations(Stream<ExtraMenuProvider> provider, FeatureRegistry featureRegistry) {
         EXTRA_DECORATIONS.clear();
-        provider.forEach(EXTRA_DECORATIONS::add);
+        PluginExtraMenuProvider.PLUGINS.clear();
+
+        provider.forEach(extra -> {
+            Plugin pl = featureRegistry.getFeatureDefinition(extra).getPlugin();
+            if (pl != null && extra.autoSubmenu()) {
+                PluginExtraMenuProvider.PLUGINS.computeIfAbsent(
+                        pl,
+                        features -> new LinkedHashSet<>()).add(extra);
+            } else {
+                EXTRA_DECORATIONS.add(extra);
+            }
+        });
+
         clean.send(null);
     }
 
+    @Feature(name = "Plugin menu provider", description = "Provides plugin extra menus")
+    public static class PluginExtraMenuProvider implements ExtraMenuProvider {
+        private static final Map<Plugin, Set<ExtraMenuProvider>> PLUGINS = new LinkedHashMap<>();
+
+        @Override
+        public Collection<JComponent> getExtraMenuItems(Main main) {
+            if (PLUGINS.isEmpty()) return Collections.emptyList();
+            List<JComponent> list = new ArrayList<>();
+            list.add(createSeparator("plugins"));
+
+            PLUGINS.forEach((plugin, features) -> {
+                list.add(createMenu(plugin.getName(),
+                    features.stream()
+                            .flatMap(f -> f.getExtraMenuItems(main).stream())));
+            });
```

Comment:
This all has the potential to create entries with no items in it, since extra menu provider could provide 0 entries depending on settings or other factors. Consider filtering that, and also adding the plugins separator only if at least something is added